x-pack/plugins/synthetics/server/routes/create_route_with_auth.ts

```diff
@@ -49,6 +49,11 @@ export const createSyntheticsRouteWithAuth = <ClientContract = unknown>(
     options,
     handler: licenseCheckHandler,
     ...rest,
+    writeAccess:
+      // if route includes an override, skip write-only access with `undefined`
+      // otherwise, if route is not a GET, require write access
+      // if route is get, use writeAccess value with `false` as default
+      writeAccessOverride === true ? undefined : method !== 'GET' ? true : writeAccess ?? false,
```

Comment:
It's a bit hard to comprehend this line tbh, extracting this into a helper function with separate `if` statements might help readability, optionally that function could also have a few unit tests. Wdyt?

Reply:
Agreed. I left the long comment to try to explain, but we can do that instead!

Reply:
Regarding run_once, I did log into your cluster and verify that the feature still worked for the admin user in the UI, so I think you are right not to worry about it. I think we are safe to merge this.

<img width="675" alt="image" src="https://github.com/elastic/kibana/assets/18429259/242c5fc2-51af-4e80-a896-0916c402534e">